WolframLanguage.sublime-syntax

```diff
@@ -174,70 +215,108 @@ contexts:
           pop: true
 
   pattern:
-    - match: '({{identifier}})\s*(:(?=[^:>=]))'
+    - match: ({{identifier}})\s*(:(?=[^:>=]))
       captures:
         1: variable.parameter.wolfram
         3: keyword.operator.Pattern.wolfram
       push:
         - meta_content_scope: meta.pattern.wolfram
         - match: \?
           scope: keyword.operator.PatternTest.wolfram
-        - match: ':(?=[^:>=])'
+        - match: :(?=[^:>=])
           scope: keyword.operator.Optional.wolfram
           pop: true
         # we should check the precedence here
         - match: (?=[:-]>|:?=|\/[;.\/]|[\]\)\},;&]|>>|<<)
           pop: true
         - include: expressions-nopattern
 
-  function:
-    - match: ({{identifier}})\s*(\[(?!\[))
-      captures:
-        1: variable.function.wolfram
-        3: punctuation.section.brackets.begin.wolfram
-      push: 
-        - meta_scope: meta.block.wolfram
-        - match: \]
-          scope: punctuation.section.brackets.end.wolfram
-          pop: true
-        - include: expressions
-    - match: ({{identifier}})(?=\s*(@\*|@{1,3}|\/[@*]))
-      scope: variable.function.wolfram
-
   variable:
-    - match: '(?:System`)?({{built_in_functions}})'
+    - match: (?:System`)?({{built_in_functions}})
       scope: variable.function
     - match: '{{identifier}}'
       scope: variable.other
     - match: (#[a-zA-Z][[:alnum:]]*|#\d*)
       scope: variable.parameter
 
-  scoping:
-    - match: (Block|Module|With|Function)\s*(\[(?!\[))
+  function:
+    # parametic functions
+    - match: ({{parametic_functions}})\s*(\[(?!\[))
+      captures:
+        1: variable.function.parametic.wolfram
+        2: punctuation.section.brackets.begin.wolfram
+      push:
+        - meta_content_scope: meta.block.wolfram
+        - match: ','
+          scope: punctuation.separator.sequence.wolfram
+          push: parameter
+        - include: meta-block
+
+    # scoping functions
+    - match: ({{scoping_functions}})\s*(\[(?!\[))
       captures:
         1: variable.function.scoping.wolfram
         2: punctuation.section.brackets.begin.wolfram
       push:
         - meta_content_scope: meta.block.wolfram
-        - match: '\{'
+        - match: \{
+          scope: punctuation.section.braces.begin.wolfram
           set: [localized, local]
-        - match: '\]'
-          scope: punctuation.section.brackets.end.wolfram
+        - include: meta-block
+
+    # functional functions
```

Comment:
I am not sure about this whole business. Can we have it in a separate PR?

Reply:
No problem

Reply:
The `meta-block` only serves as a abstraction for some of the contexts.